sanic/blueprints.py

```diff
@@ -449,16 +449,11 @@ def event(self, event: str, timeout: Optional[Union[int, float]] = None):
 
     @staticmethod
     def _extract_value(*values):
-        value = values[-1]
-        for v in values:
-            if v is not None:
-                value = v
-                break
-        return value
+        return next((v for v in values if v is not None), values[-1])
```

Comment:
@ahopkins is this to really just return the first value found that is not None? Just want to make sure I understand what you were after here. I'm okay with switching to the generator as it's a parity change, but I want to verify that's what you were really after.

Reply:
yea, it returns the first value found that is not None

Reply:
@Mng-dev-ai yes, but I want to make sure that you didn't uncover a bug, @ahopkins will probably look tomorrow.

Reply:
The logic seems fine. We probably could improve DRY a little by using `_extract_value` with the `version_prefix` and just abstract the conditional a bit since they are essentially doing the same thing, and it would keep the priority setting using a consistent methodology.

However, TBH, I am not sure how this is an improvement and what this change adds. Even running 10M iterations over it, I only had to run my benchmarks twice to get inconsistent results. I am not sure that I would agree that a generator adds any performance here.
![image](https://user-images.githubusercontent.com/166269/164973604-cd352a68-abab-426a-815f-be171dbb9e34.png)

We are only looping through 2 or three items, not a very large list. And, the values already exist in memory so using a generator does not reduce the footprint. Am I missing something else?

Reply:
Didn't know that the array has only a few items so as u said the difference won't be noticed